internal/transport/transport.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/mongodb-forks/digest"
+	atlasauth "go.mongodb.org/atlas/auth"
+)
+
+const (
+	telemetryTimeout      = 1 * time.Second
+	timeout               = 5 * time.Second
+	keepAlive             = 30 * time.Second
+	maxIdleConns          = 5
+	maxIdleConnsPerHost   = 4
+	idleConnTimeout       = 30 * time.Second
+	expectContinueTimeout = 1 * time.Second
+)
+
+func Default() *http.Transport {
+	return newTransport(timeout)
+}
+
+func Telemetry() *http.Transport {
+	return newTransport(telemetryTimeout)
+}
```

Comment:
could these be singleton? the difference right now with this approach is that there use to be a single instance of transport while now you'll create multiple ones